File: leetcode/algorithms/561_array_partition/main.py
from typing import List


class ArrayPartition:
    def arrayPairSum(self, nums: List[int]) -> int:
        result = 0

        nums.sort()  # or nums = sorted(nums)

        for i in range(len(nums)):
            if i % 2 == 0:
                result += nums[i]

        return result


    # Solution
    def solution(self, nums: List[int]) -> int:
        return sum(sorted(nums)[::2])

File: leetcode/algorithms/561_array_partition/main.dart
import 'dart:math';

class ArrayPartition {
  int arrayPairSum(List<int> nums) {
    nums.sort();
    int result = 0;
    for (int i = 0; i < nums.length; i += 2) {
      result += nums[i];
    }
    return result;
  }

  // Solution
  int solution(List<int> nums) {
    nums.sort((a, b) => b.compareTo(a));
    int sum = 0;
    for (int i = 0; i < nums.length; i++) {
      if (i % 2 != 0) {
        sum += min(nums[i], nums[i - 1]);
      }
    }
    return sum;
  }
}
